From 67fad6f56906679a713ea2ee263d082487cdd8e6 Mon Sep 17 00:00:00 2001
From: Mark Thomas <markt@apache.org>
Date: Thu, 19 Feb 2009 15:57:25 +0000
Subject: [PATCH] Fix DBCP-215. Make sure initialSize connections are placed in
 the pool are created when the pool is initialized tather than on first use.
 Patch suggested by psteitz

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/dbcp/trunk@745908 13f79535-47bb-0310-9956-ffa450edef68
---
 src/java/org/apache/commons/dbcp/BasicDataSourceFactory.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/java/org/apache/commons/dbcp/BasicDataSourceFactory.java b/src/java/org/apache/commons/dbcp/BasicDataSourceFactory.java
index 9bdb646710..cfe41243fa 100644
--- a/src/java/org/apache/commons/dbcp/BasicDataSourceFactory.java
+++ b/src/java/org/apache/commons/dbcp/BasicDataSourceFactory.java
@@ -322,6 +322,10 @@ else if ("SERIALIZABLE".equalsIgnoreCase(value)) {
           }
         }
 
+        // DBCP-215
+        // Trick to make sure that initialSize connections are created
+        dataSource.getLogWriter();
+
         // Return the configured DataSource instance
         return dataSource;
     }
